__tests__/server/middleware/pwa/config.spec.js

```diff
@@ -0,0 +1,192 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import {
+  resetPWAConfig,
+  getPWAConfig,
+  getClientPWAConfig,
+  setPWAConfig,
+  validatePWAConfig,
+  configurePWA,
+} from '../../../../src/server/middleware/pwa/config';
+
+jest.mock('fs', () => ({
+  readFileSync: (filePath) => ({ toString: () => (filePath.endsWith('noop.js') ? '[service-worker-noop-script]' : '[service-worker-script]') }),
+}));
+
+describe('pwa configuration', () => {
+  test('exports functions as default', () => {
+    expect.assertions(6);
+
+    expect(resetPWAConfig).toBeInstanceOf(Function);
+    expect(getPWAConfig).toBeInstanceOf(Function);
+    expect(getClientPWAConfig).toBeInstanceOf(Function);
+    expect(setPWAConfig).toBeInstanceOf(Function);
+    expect(validatePWAConfig).toBeInstanceOf(Function);
+    expect(configurePWA).toBeInstanceOf(Function);
+  });
+
+  test('getters return default state', () => {
+    expect.assertions(2);
+
+    expect(getPWAConfig()).toMatchObject({
+      enabled: false,
+      escapeHatch: false,
+      noop: false,
+    });
+    expect(getClientPWAConfig()).toMatchObject({
+      enabled: false,
+      scope: null,
+      manifest: null,
+      scriptUrl: null,
+    });
+  });
+
+  test('setting service worker configuration', () => {
+    expect.assertions(3);
+
+    expect(setPWAConfig({
+      enabled: true,
+    })).toMatchObject({
+      enabled: true,
+      escapeHatch: false,
+      noop: false,
+    });
+    expect(setPWAConfig({
+      escapeHatch: true,
+    })).toMatchObject({
+      enabled: false,
+      escapeHatch: true,
+      noop: false,
+    });
+    expect(setPWAConfig({
+      noop: true,
+    })).toMatchObject({
+      enabled: false,
+      escapeHatch: false,
+      noop: true,
+    });
+  });
+
+  describe('validation', () => {
+    beforeEach(() => {
+      jest.clearAllMocks();
+    });
+    const { error, warn } = console;
+    beforeAll(() => {
+      console.warn = jest.fn();
+      console.error = jest.fn();
+    });
+    afterAll(() => {
+      console.warn = warn;
+      console.error = error;
+    });
+
+    test('invalid configuration object informs the user', () => {
+      expect.assertions(9);
+
+      expect(validatePWAConfig(null)).toEqual(null);
+      expect(console.error).toHaveBeenCalledTimes(1);
+      expect(console.error).toHaveBeenCalledWith('invalid config given to service worker (expected "object")');
```

Comment:
this assertion is the same every time. unless you reset the mock between calls, this will always be true

Reply:
I've removed the extra checks.